app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,31 @@
+class SessionsController < ApplicationController
+  include SessionsHelper
+
+  def new
+  end
+
+  def create
+    @user = User.find_by(email: params[:session][:email].downcase)
+    if @user && @user.authenticate(params[:session][:password])
```

Comment:
```suggestion
     if can_authenticate?(user) # extract logic into well-named method
```
"Ditto for optimizing for readability, better to create well-named abstractions and leave it up to the reader to decide if they want to dig into implementation details, that includes yourself 6 months from now when you've forgotten about this code"

by @c13e

Reply:
Thanks, learnt Alot through these comments  appreciate it @LindoDev @c13e